tests/bip39.rs

```diff
@@ -61,54 +66,24 @@ fn test_wordlist_codec() {
         let ws = choose_wordlist();
 
         let ms = wordlist::encode(&data, ws).unwrap();
-        assert_eq!(wordlist::decode(&ms, ws).unwrap(), data);
-        assert_eq!(wordlist::verify(&ms, ws), Ok(()));
-    }
-}
-
-#[test]
-fn test_mnemonic_phrase_when_separator_is_repeated() {
-    let test_cases = &[
-        // U+3000 separator
-        ("　", true),
-        // whitespace(U+0020) is also allowed as a separator, because U+3000 is normalized to the whitespace
-        (" ", true),
-        (" 　", false),
-        ("  ", false),
-        ("　 ", false),
-        ("　 　", false),
-    ];
-
-    for case in test_cases {
-        let mnemonic_phrase = format!("あいこくしん{}あいこくしん　あいこくしん　あいこくしん　あいこくしん　あいこくしん　あいこくしん　あいこくしん　あいこくしん　あいこくしん　あいこくしん　あおぞら", case.0);
-        assert_eq!(
-            wordlist::decode(&mnemonic_phrase, &wordlist::JAPANESE).is_ok(),
-            case.1,
-            "{}",
-            mnemonic_phrase
-        );
-        assert_eq!(
-            wordlist::verify(&mnemonic_phrase, &wordlist::JAPANESE).is_ok(),
-            case.1,
-            "{}",
-            mnemonic_phrase
-        );
+        assert_eq!(*wordlist::decode((&ms).into(), ws).unwrap(), data);
+        assert_eq!(wordlist::verify((&ms).into(), ws), Ok(()));
     }
 }
 
 #[test]
 fn test_mnemonic_phrase_additional_whitespace() {
     // additional whitespace at the beginning
     assert!(
-        wordlist::decode(" sand luggage rack used middle crater deal scare high ring swim fish use then video visa can foot clog base quality all elephant retreat", &wordlist::ENGLISH).is_err(),
+        wordlist::decode(" sand luggage rack used middle crater deal scare high ring swim fish use then video visa can foot clog base quality all elephant retreat".try_into().unwrap(), &wordlist::ENGLISH).is_err(),
```

Comment:
I know it wasn't there, but can you test that the error is the expected one?

Reply:
Good point. I didn't touch this subject in the PR. We should probably test for different error cases.

Reply:
Done.